File: examples/python/openai_integration.py
from openai import OpenAI
from openai.types.responses.tool_param import Mcp

client = OpenAI(api_key="your_openai_key")

mcp_tool = Mcp(
    type="mcp",
    server_label="youtube2text",
    server_url="https://api.youtube2text.org/mcp",
    require_approval="never",
    headers={"Authorization": "Bearer your_yt2text_key"}
)

response = client.responses.create(
    model="gpt-4.1",
    tools=[mcp_tool],
    input="Transcribe and analyze: https://www.youtube.com/watch?v=example"
)

print(response.output_text)

File: examples/typescript/openai_integration.ts
import OpenAI from "openai";

const openai = new OpenAI(); // reads OPENAI_API_KEY from the environment

async function main(): Promise<void> {
  const response = await openai.responses.create({
    model: "gpt-5",
    tools: [
      {
        type: "mcp",
        server_label: "youtube2text",
        server_url: "https://youtube2text.org/mcp",
        headers: { "x-api-key": "your_yt2text_key" },
        require_approval: "never",
      },
    ],
    input: "Transcribe and analyze: https://www.youtube.com/watch?v=example",
  });

  console.log(response.output_text);
}

main();
